library/network_connections.py

```diff
@@ -1025,9 +1035,20 @@ def connection_create(self, connections, idx, connection_current=None):
                 s_ip6.set_property(
                     NM.SETTING_IP_CONFIG_ROUTE_METRIC, ip["route_metric6"]
                 )
-            for d in ip["dns"]:
-                if d["family"] == socket.AF_INET6:
-                    s_ip6.add_dns(d["address"])
+            for nameserver in ip["dns"]:
+                if nameserver["family"] != socket.AF_INET6:
+                    continue
```

Comment:
Why are you using the negation with continue here? It seems to me to be easier to understand to use a non-negated if statement here without continue.

Reply:
sure.

The "continue" is not too bad IMO, because then you walk through the loop and abort it early (when a certain condition is satisfied). The alternative leads to deeper nested code/block, which IMO hurt readability.

Of course, not a big issue either way. But that was my reasoning.

Reply:
If the check for disabled IPv6 is done in the argument validator (which might be the actual place to do this), then the for loop is just one statement, though.

Reply:
True!! When I originally suggested the snippet, I thought there would be more inside the loop.

Reply:
I changed into non-negated if statement without `continue`:
```
            for nameserver in ip["dns"]:
                if nameserver["family"] == socket.AF_INET6:
                    s_ip6.add_dns(nameserver["address"])
```